dropshot/src/http_util.rs

```diff
@@ -35,9 +36,11 @@ where
     // TODO better understand pin_mut!()
     // TODO do we need to use saturating_add() here?
     let mut nbytesread: usize = 0;
-    while let Some(maybebuf) = body.data().await {
-        let buf = maybebuf?;
-        nbytesread += buf.len();
+    while let Some(maybefr) = body.frame().await {
+        let fr = maybefr?;
+        if let Ok(buf) = fr.into_data() {
+            nbytesread += buf.len();
+        }
     }
 
     // TODO-correctness why does the is_end_stream() assertion fail?
```

Comment:
I think that the commented-out assertion here was _probably_ failing because, after discarding all the `DATA` frames, an HTTP/2 body was terminated by a `TRAILERS`, so the stream was not EOF. I wonder if this assertion can be put back, now that the new implementation will also read and discard any `TRAILERS` frame that's sent as well as the `DATA` frames.